lib/WebService/Client.pm

```diff
@@ -103,6 +100,26 @@ sub log {
     $self->logger->$log_method($msg);
 }
 
+sub prepare_response {
+    my ($self, $res, $req, $num_attempts) = @_;
+    Moo::Role->apply_roles_to_object($res, 'HTTP::Response::Stringable');
+    return;
+}
+
+sub parse_response {
+    my ($self, $res, $req, $num_attempts) = @_;
+
+    return undef if $req->method eq 'GET' && $res->code =~ qr/ 404 | 410 /x;
+    die $res if not $res->is_success;
+    return decode_json($res->content) if $res->content;
+    return 1;
+}
```

Comment:
I think this default behavior is ok in most cases, but consumers might want to customize parsing (e.g. using XML instead of JSON) or error handing (e.g. throwing a custom error).